core/src/components/range/range.tsx

```diff
@@ -629,7 +636,7 @@ Developers can dismiss this warning by removing their usage of the "legacy" prop
               'label-text-wrapper-hidden': !this.hasLabel,
             }}
           >
-            <slot name="label"></slot>
+            {label !== undefined ? <div class="label-text">{label}</div> : <slot name="label"></slot>}
```

Comment:
Should we be conditionally rendering the `<slot>` or should we be conditionally hiding/showing the slot?

Reply:
I went with conditional rendering for performance reasons; there's little reason to render the slot content if it's just going to be hidden anyway. Do you think it should be hidden instead?

Reply:
Let's keep it the way you have it. We should be able to easily change the behavior to hide the slot if the current approach causes problems.